src/route_calc.c

```diff
@@ -52,12 +52,41 @@ static void TravelTimeForPath(gedict_t *m, int i)
 		player_speed = sv_maxspeed * (1 + max(0, DotProduct(distance, hor_distance)));
 
 		// FIXME: RJ time is guideline, but we can do better than this?
-		m->fb.paths[i].time = 100000;
+		m->fb.paths[i].time = 1000000;
 		m->fb.paths[i].rj_time = (total_distance / player_speed);
 
 		return;
 	}
 
+	if (m->fb.paths[i].flags & HOOK)
+	{
+		vec3_t m_P_pos;
+		vec3_t distance;
+		vec3_t hor_distance;
+		float player_speed;
+		float total_distance;
+
+		VectorAdd(m_P->s.v.absmin, m_P->s.v.view_ofs, m_P_pos);
+		VectorSubtract(m_P_pos, m_pos, distance);
+		VectorCopy(distance, hor_distance);
+		total_distance = VectorNormalize(distance);
+		VectorNormalize(hor_distance);
+		player_speed = sv_maxspeed * (1 + max(0, DotProduct(distance, hor_distance)));
+
+		// TODO hiipe - work out time correctly!
+		m->fb.paths[i].time = /*1000000*/(total_distance / player_speed);
+		//m->fb.paths[i].hook_time = (total_distance / player_speed);
```

Comment:
Does it work at all? Needs more work? Otherwise as with other stuff, file a GH issue with context and possibly half baked ideas. Perhaps someone else happens to read it and implement it for you compared to hiding it in the code.